### ISSUE TITLE
Jest executor does not cope with dependencies that only provide esmodule formatted js

### ISSUE THREAD

**Issue by jgbright** [nrwl/nx#10605]
TL;DR: I suspect this is a dupe of other issues, but I have assembled a minimal repro repo that should be a good point of reference for discussion.  https://github.com/jgbright/nx-jest-ky

## Current Behavior
Jest tests fail to run if they reference an npm package built only with esmodule formatted contents.

## Expected Behavior
I expected the jest tests to compile and run.

I don't know if this is a regression.

## Steps to Reproduce
- Generate an nx workspace with the react template.
- Generate an nx library.
- In the lib, use an npm package that happens to only have esmodules support (like ky).

For more detail, see minimal github repo that shows repro steps.
https://github.com/jgbright/nx-jest-ky

### Failure Logs
Transcripts of running various nx commands available here:  https://github.com/jgbright/nx-jest-ky/blob/main/README.md

### Environment
```
➜ nx report

 >  NX   Running global Nx CLI with PNPM may have issues.

   Prefer to use "pnpm" (https://pnpm.io/cli/exec) to execute commands in this workspace.
    TIP  create a shortcut such as: doskey pnx=pnpm nx -- $*
   


 >  NX   Report complete - copy this into the issue template

   Node : 18.2.0
   OS   : win32 x64
   pnpm : 6.32.17
   
   nx : 14.1.9
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.1.9
   @nrwl/detox : Not Found
   @nrwl/devkit : Not Found
   @nrwl/eslint-plugin-nx : 14.1.9
   @nrwl/express : Not Found
   @nrwl/jest : 14.1.9
   @nrwl/js : Not Found
   @nrwl/linter : 14.1.9
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.1.9
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : 14.1.9
   @nrwl/workspace : 14.1.9
   typescript : 4.6.4
   ---------------------------------------
   Community plugins:

 C:\Repos\nx-jest-ky :: main ≡                                                                                 |  18.2.0 |  100  | 17:52:47  
➜
```

**Comment by jgbright** [comment]
I have seen workarounds in other issues, but I couldn't get any of them to work in my project.  It would be great to see a workaround that works for this very simple repo.

**Comment by barbados-clemens** [comment]
You'll have to transform esm code or point to a common js version with a moduleNameMapper/custom resolver (careful to not remove the nrwl resolver functionally. i.e. extend our resolver vs replacing it)

though this looks like it might be a duplicate of #8323. Can you try some of those steps defined in that issue, which is basically the babel-jest transformer isn't transforming
